### PR TITLE
[assertj] Add VersionRangeAssert

### PR DESCRIPTION
**Pull request by kriegfrj**
Signed-off-by: Fr Jeremy Krieg <fr.jkrieg@greekwelfaresa.org.au>

### CODE REVIEW

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/versionrange/VersionBoundAssert.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * limitations under the License.
+ */
+
+package org.osgi.test.assertj.versionrange;
+
+import org.assertj.core.api.InstanceOfAssertFactory;
+import org.osgi.framework.Version;
+
+public class VersionBoundAssert extends AbstractVersionBoundAssert<VersionBoundAssert, Version> {
+	public static InstanceOfAssertFactory<Version, VersionBoundAssert>	OPEN_VERSION	= new InstanceOfAssertFactory<>(
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/versionrange/VersionBoundAssert.java:23]: OPEN_VERSION and CLOSED_VERSION need to be final.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/versionrange/AbstractVersionRangeAssert.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+		if (!intersection
+			.isEmpty()) {
+			failWithMessage(
+				"%nExpecting version range%n <%s>%nto not intersect any of%n <%s>%nbut it has intersection%n <%s>",
+				actual, Arrays.toString(versions), intersection);
+		}
+		return myself;
+	}
+
+	// public AbstractIntegerAssert<?> hasMinorThat() {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/versionrange/AbstractVersionRangeAssert.java:204]: I don't think the commented section should be in the PR. There is some overlap (isEmpty) with the code.

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/versionrange/VersionBoundAssert.java (hunk 1 above)
Comment (kriegfrj) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/versionrange/VersionBoundAssert.java:23]: Done!

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/versionrange/AbstractVersionRangeAssert.java (hunk 2 above)
Comment (kriegfrj) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/versionrange/AbstractVersionRangeAssert.java:204]: Oops. Done.

**Review by bjhargrave — APPROVED** [review]

**Pull request merged**